applications/GeoMechanicsApplication/custom_retention/retention_law.h

```diff
@@ -162,70 +123,32 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) RetentionLaw
          * Returns the reference or the value of a specified variable: returns the value of the parameter, only non const values can be modified
          */
 
-        const double &GetVolumetricStrain() { return mVolumetricStrain; }
-        const double &GetMeanStress()       { return mMeanStress;       }
-        const double &GetFluidPressure()    { return mFluidPressure;    }
-        const double &GetTemperature()      { return mTemperature;      }
+        double GetVolumetricStrain() const { return mVolumetricStrain; }
+        double GetMeanStress()       const { return mMeanStress;       }
+        double GetFluidPressure()    const { return mFluidPressure;    }
+        double GetTemperature()      const { return mTemperature;      }
 
         const ProcessInfo &GetProcessInfo()
         {
-            KRATOS_DEBUG_ERROR_IF_NOT(IsSetProcessInfo()) << "ProcessInfo is not set!" << std::endl;
+            KRATOS_DEBUG_ERROR_IF_NOT(mpCurrentProcessInfo != NULL) << "ProcessInfo is not set!" << std::endl;
```

Comment:
In line with "ES.47: Use `nullptr` rater than `0` or `NULL`" of the C++ Core Guidelines, I would replace `NULL` by `nullptr`. Furthermore, I would try to avoid the double negation (`(...)_IF_NOT` and `!=`) since that is somewhat harder to read. I would prefer this:
```suggestion
            KRATOS_DEBUG_ERROR_IF(mpCurrentProcessInfo == nullptr) << "ProcessInfo is not set!" << std::endl;
```